pkg/v1/webhooks/certificates.go

```diff
@@ -122,3 +136,32 @@ func ValidateTLSSecret(tlsSecret *corev1.Secret, certGracePeriod time.Duration)
 	}
 	return nil
 }
+
+// write secret to cluster
+func updateOrCreateTLSSecret(ctx context.Context, clusterClient client.Client, tlsSecret *corev1.Secret) error {
```

Comment:
I would rename clusterClient to k8sclient, otherwise it can be confused with a remote clusterClient

Reply:
done